src/components/shared/table/Table.tsx

```diff
@@ -21,11 +24,12 @@ function TableComponent<T extends { id?: number }>({
   className,
   renderActions,
 }: TableProps<T>): JSX.Element {
+  const { translate } = useTranslate();
   if (data.length === 0) {
     return (
       <div className="no-data">
         <img src={EmptyIcon} />
-        <p>No data available</p>
+        <p>{translate('No data available')}</p>
       </div>
     );
   }
```

Comment:
The is no need to catch when we don't have any data since you handled this in your table logic here:
```TSX
<tbody>
          {data.length === 0 ? (
            <tr>
              <td className="no-data" colSpan={columns.length + 1}>
                No data available
              </td>
            </tr>
```